lib/wallet/path.js

```diff
@@ -212,14 +213,23 @@ class Path extends bio.Struct {
 
   /**
    * Convert path object to string derivation path.
+   * @param {String|Network?} network - Network type.
    * @returns {String}
    */
 
-  toPath() {
+  toPath(network) {
     if (this.keyType !== Path.types.HD)
       return null;
 
-    return `m/${this.account}'/${this.branch}/${this.index}`;
```

Comment:
I don't love how this function returns a different depth path depending on `network` being defined, but its a low touch change that works. While the only `purpose` right now is `44`, I can think of usecases for different purpose fields like rolling over the keys that control a name in a deterministic and auditable way that leaks no additional information about other transactions sent.  Thats definitely out of scope for this PR, but I think that its good to keep in mind. Also keep in mind the `PSBT_OUT_BIP32_DERIVATION` field in a [PSBT](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki), where `prefix` would be replaced with the fingerprint of `M` (depth 0 public key). These suggestions are more immediately applicable to bcoin, but could also be useful to hsd

Reply:
> I don't love how this function returns a different depth path depending on network being defined, but its a low touch change that works.

I agree. I originally created a new method: `toFullPath` but I did not want to completely change the behavior of the Path object (particular `getJSON`).

Re: support for alternative purposes -- I think we should handle this by either a redesign to the Path object, or specific path objects for each specification of HD. The current object assumes BIP44 style derivation and makes assumptions about the purpose and coin type indices.

Reply:
I believe most of the codebase in bcoin assumes `network=main` if nothing is passed. Wallet always has information about network.

Reply:
@nodar-chkuaselidze @tynes the issue here is about backward compatibility. As written, this change tries to introduce the ability to include the `purpose` (defaults to 44) and `coin_type` in the `derivation` attribute without breaking any current usage of `toPath`. Do you think it would be better to update `toPath` to always return a full, BIP44 path? If so, using `network=main` as a default makes a lot of sense. If we care about backward compatibility (more of an issue when we port this change to `bcoin`), then I do not believe we will want to use a default for `network`.

Reply:
>  Do you think it would be better to update toPath to always return a full, BIP44 path?

I think that this is ok for now but we should really figure out the right way to upgrade so that arbitrary bip44 paths are supported